### PR TITLE
Add tests to check libvirt hooks.

### PR DESCRIPTION
**Pull request by rbian**

### CODE REVIEW

File: libvirt/tests/src/libvirt_hooks.py [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+from virttest import utils_libvirtd
+from virttest.utils_test import libvirt
+from virttest.libvirt_xml import vm_xml
+
+
+def run(test, params, env):
+    """
+    Test interafce xml options.
+
+    1.Prepare test environment,destroy or suspend a VM.
```
Comment (will-Do) [inline libvirt/tests/src/libvirt_hooks.py]: s/,/, /

File: libvirt/tests/src/libvirt_hooks.py [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+        elif basic_test:
+            prepare_hook_file(hook_script %
+                              (vm_name, hook_log))
+        ret = virsh.save(vm_name, save_file, **virsh_dargs)
+        libvirt.check_exit_status(ret)
+        if domainxml_test:
+            disk_src_save = vm.get_first_disk_devices()['source']
+            if disk_src != disk_src_save:
+                raise error.TestFail("Failed to check hooks for"
+                                     " save operation")
```
Comment (will-Do) [inline libvirt/tests/src/libvirt_hooks.py]: After raise TestFail, the cleanup steps will not execute, so as the ``` managedsave_hook()``` function.

File: libvirt/tests/src/libvirt_hooks.py [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+        """
+        Check attach hooks.
+        """
+        # Start a domain with qemu command.
+        disk_src = vm.get_first_disk_devices()['source']
+        vm_test = "foo"
+        prepare_hook_file(hook_script %
+                          (vm_test, hook_log))
+        qemu_cmd = ("/usr/libexec/qemu-kvm -cdrom %s -monitor "
+                    "unix:/tmp/demo,server,nowait -name %s"
```
Comment (will-Do) [inline libvirt/tests/src/libvirt_hooks.py]: The qeme-kvm binary may a different path or name on different system, so maybe let 'qemu_cmd' configurable is better

**Comment by rbian** [comment]
Test results:
./scripts/avocado run --vt-type libvirt libvirt_hooks
JOB ID     : 7a6c452737e0bb812e7d93745419084b822e0e83
JOB LOG    : /tmp/avocado-test-40/avocado-vt/job-results/job-2015-09-24T18.21-7a6c452/job.log
TESTS      : 12
 (1/12) type_specific.io-github-autotest-libvirt.libvirt_hooks.start_stop:  PASS (14.91 s)
 (2/12) type_specific.io-github-autotest-libvirt.libvirt_hooks.libvirtd_t:  PASS (91.84 s)
 (3/12) type_specific.io-github-autotest-libvirt.libvirt_hooks.save_restore.basic_test:  PASS (88.17 s)
 (4/12) type_specific.io-github-autotest-libvirt.libvirt_hooks.save_restore.domainxml_test:  PASS (147.30 s)
 (5/12) type_specific.io-github-autotest-libvirt.libvirt_hooks.managedsave_t.basic_test:  PASS (93.10 s)
 (6/12) type_specific.io-github-autotest-libvirt.libvirt_hooks.managedsave_t.domainxml_test:  PASS (124.52 s)
 (7/12) type_specific.io-github-autotest-libvirt.libvirt_hooks.virsh_attach:  PASS (15.78 s)
 (8/12) type_specific.io-github-autotest-libvirt.libvirt_hooks.daemon_t.exit_0:  PASS (20.66 s)
 (9/12) type_specific.io-github-autotest-libvirt.libvirt_hooks.daemon_t.exit_1:  PASS (20.64 s)
 (10/12) type_specific.io-github-autotest-libvirt.libvirt_hooks.network_t:  PASS (32.23 s)
 (11/12) type_specific.io-github-autotest-libvirt.libvirt_hooks.error_test.start_error:  PASS (1.41 s)
 (12/12) type_specific.io-github-autotest-libvirt.libvirt_hooks.scale_test:  PASS (390.80 s)
RESULTS    : PASS 12 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0

File: libvirt/tests/src/libvirt_hooks.py (hunk 2 above)
Comment (rbian) [inline libvirt/tests/src/libvirt_hooks.py]: No, "finally" section will always be executed.

**Comment by will-Do** [comment]
Thanks @rbian 
I'm OK for this test, so ACKed.

**Pull request merged**